tgui/packages/tgui/interfaces/TEG.js

```diff
@@ -0,0 +1,135 @@
+/**
+ * @file
+ * @copyright 2020
+ * @author Sovexe (https://github.com/Sovexe)
+ * @license ISC
+ */
+
+import { useBackend } from '../backend';
+import { Box, Chart, LabeledList, Section, Divider } from '../components';
+import { formatPower, formatSiUnit } from '../format';
+import { Window } from '../layouts';
+
+export const TEG = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    output,
+    history,
+    hotCircStatus,
+    hotInletTemp,
+    hotOutletTemp,
+    hotInletPres,
+    hotOutletPres,
+    coldCircStatus,
+    coldInletTemp,
+    coldOutletTemp,
+    coldInletPres,
+    coldOutletPres,
+  } = data;
+  const historyData = history.map((value, i) => [i, value]);
+  const historyMax = Math.max(...history);
+  return (
+    <Window
+      height="520"
+      width="300" >
+      <Window.Content>
+        <Section title="Status">
+          <LabeledList>
+            <LabeledList.Item label="Output History" />
+          </LabeledList>
+          <Chart.Line
+            height="5em"
+            data={historyData}
+            rangeX={[0, historyData.length - 1]}
+            rangeY={[0, historyMax]}
+            strokeColor="rgba(1, 184, 170, 1)"
+            fillColor="rgba(1, 184, 170, 0.25)" />
+          <Divider />
+          <LabeledList>
+            <LabeledList.Item
+              label="Energy Output"
+              textAlign="right" >
+              {formatPower(output)}
+            </LabeledList.Item>
+            <LabeledList.Item
+              label="Hot Gas Circulator"
+              textAlign="right" >
+              <Box color={(hotCircStatus && hotInletTemp && 'good')
+                  || (hotCircStatus && 'average')
+                  || 'bad'} >
+                {(hotCircStatus && hotInletTemp && 'OK')
+                  || (hotCircStatus && 'Idle')
+                  || 'ERROR'}
+              </Box>
+            </LabeledList.Item>
+            <LabeledList.Item
+              label="Cold Gas Circulator"
+              textAlign="right" >
+              <Box color={(coldCircStatus && coldInletTemp && 'good')
+                  || (coldCircStatus && 'average')
+                  || 'bad'} >
+                {(coldCircStatus && coldInletTemp && 'OK')
+                  || (coldCircStatus && 'Idle')
+                  || 'ERROR'}
+              </Box>
+            </LabeledList.Item>
+          </LabeledList>
+        </Section>
+        <Section title="Hot Loop">
+          <LabeledList>
+            <LabeledList.Item
+              label="Inlet Temp"
+              textAlign="right" >
+              {(hotInletTemp >= 1000 && hotInletTemp.toExponential(3) + ' K')
+                || hotInletTemp + ' K'}
```

Comment:
```suggestion
              {`${hotInletTemp >= 1000 ? hotInletTemp.toExponential(3) : hotInletTemp} K`}
```

Reply:
Added suggestions to the four usages in this file, but as they're all basically the same you could also do a quick format function and pass in just the relevant variable, too, e.g.
```
const formatTemperature = temperature => `${temp >= 1000 ? temp.toExponential(3) : temp} K`;
// ...
// usage for here
{formatTemperature(hotInletTemp)}
```
Good rule of thumb from me for UI stuff is if you're doing the same thing three or more times, consider making a function/component (as relevant) for it.